### PR TITLE
Allow system table to be set as default database

### LINKED ISSUE #7146: Unknown database 'information_schema'
#### Reproduction Steps

When trying to run `skeema init` against vtgate, I get the following error:

```
ERROR 1049 (42000): vtgate: http://user-ThinkPad-X1-Carbon-7th:9997/: Unknown database 'information_schema' (errno 1049) (sqlstate 42000)
```

This is because vtgate is looking for a keyspace in the vschema named `information_schema`:
```
https://github.com/vitessio/vitess/blob/2eb5d64406a438467f17710e822d2852b37526ba/go/vt/vtgate/vcursor_impl.go#L454

```

and the fact the skeema wants to connect directly to `information_schema`: 
```
https://github.com/skeema/skeema/blob/c3469f0286dd269ac43158befc39927f9fc061a8/vendor/github.com/skeema/tengo/instance.go#L783
```

The minimal way to reproduce this is:
```
mysql --host 127.0.0.1 --port 10000 information_schema
```
where `10000` is the port vtgate is listening on.

### PR DESCRIPTION
**Pull request by harshit-gangal**
Signed-off-by: Harshit Gangal <harshit@planetscale.com>

## Backport
NO

## Status
**READY**

## Description
This PR allows setting system schema as default database name on connect or with `use` statement. 

## Related Issue(s)
List related PRs against other branches:
Fixes #7146 

## Impacted Areas in Vitess
List general components of the application that this PR will affect:

- [X]  Query Serving

### CODE REVIEW

File: go/vt/sqlparser/ast_rewriting.go [hunk 1]
```diff
@@ -110,7 +112,7 @@ const (
 )
 
 func (er *expressionRewriter) rewriteAliasedExpr(node *AliasedExpr) (*BindVarNeeds, error) {
-	inner := newExpressionRewriter()
+	inner := newExpressionRewriter("")
```
Comment (systay) [inline go/vt/sqlparser/ast_rewriting.go:115]: should this string be `er.keyspace` instead of ""?

Comment (harshit-gangal) [reply go/vt/sqlparser/ast_rewriting.go:115]: I see no harm. Fixed.

File: go/vt/sqlparser/ast_rewriting_test.go [hunk 2]
```diff
@@ -199,3 +199,49 @@ func TestRewrites(in *testing.T) {
...
+			require.NoError(err)
+
+			result, err := RewriteAST(stmt, "sys")
+			require.NoError(err)
+
+			expected, err := Parse(tc.expected)
+			require.NoError(err, "test expectation does not parse [%s]", tc.expected)
+
+			s := String(expected)
+			assert := assert.New(t)
```
Comment (systay) [inline go/vt/sqlparser/ast_rewriting_test.go:243]: nit: meh... if you are using a single assertion, not much use creating this struct, is there?

**Review by systay — APPROVED** [review]
Really nice!

**Pull request merged**